File: backend/src/main/java/com/github/fauu/natrank/service/TeamService.java
package com.github.fauu.natrank.service;

import com.github.fauu.natrank.model.entity.Team;
import com.github.fauu.natrank.model.entity.TeamRank;
import com.github.fauu.natrank.model.entity.TeamRating;
import org.springframework.dao.DataAccessException;

import java.util.List;

public interface TeamService {

  Team findById(int id) throws DataAccessException;

  Team findByName(String name) throws DataAccessException;

  List<TeamRank> findRanksByName(String name) throws DataAccessException;

  List<TeamRating> findRatingsByName(String name) throws DataAccessException;

}

File: frontend-old/service/teamService.js

angular.module('natrank')
.service('teamService', ['$http', 'NATRANK_CONFIG',
function($http, NATRANK_CONFIG) {
  var urlBase = NATRANK_CONFIG.baseUrl + '/teams';

  this.findByName = function(name) {
    return $http.get(urlBase + '/' + name);
  };

  this.findRanksByName = function(name) {
    return $http.get(urlBase + '/' + name + '/ranks');
  };

  this.findRatingsByName = function(name) {
    return $http.get(urlBase + '/' + name + '/ratings');
  };
}]);
